packages/SwingSet/src/vats/network/network.js

```diff
@@ -371,25 +372,24 @@ export function makeNetworkProtocol(protocolHandler) {
     bind,
     async inbound(listenAddr, remoteAddr) {
       let lastFailure = Error(`No listeners for ${listenAddr}`);
-      for (const listenPrefix of getPrefixes(listenAddr)) {
+      for await (const listenPrefix of getPrefixes(listenAddr)) {
         if (!listening.has(listenPrefix)) {
-          // eslint-disable-next-line no-continue
           continue;
         }
         const [port, listener] = listening.get(listenPrefix);
         let localAddr;
-        try {
+        await (async () => {
           // See if our protocol is willing to receive this connection.
-          // eslint-disable-next-line no-await-in-loop
           const localInstance = await E(protocolHandler)
             .onInstantiate(port, listenPrefix, remoteAddr, protocolHandler)
             .catch(rethrowUnlessMissing);
           localAddr = localInstance
             ? `${listenAddr}/${localInstance}`
             : listenAddr;
-        } catch (e) {
+        })().catch(e => {
```

Comment:
I'm pleased to see the direct readability penalty is so low. A small price for more safety. I have no problem at all recommending this.

OTOH, these examples show an additional indirect readability penalty of not being able to do a non-local exit directly from catch callback; needing to use state variables instead. Still, this seems like a big win. Thanks!

No change suggested.